libraries/chain/block_state_legacy.cpp

```diff
@@ -84,4 +84,26 @@ namespace eosio::chain {
    {
    }
 
+   block_num_type block_state_legacy::irreversible_blocknum() const {
+      if (std::optional<block_num_type> gen_block_num = savanna_genesis_block_num()) {
+         return std::min<block_num_type>(*gen_block_num, dpos_irreversible_blocknum);
```

Comment:
shouldn't it be `std::max()`?

Reply:
We don't want LIB in legacy to advance past savanna genesis block.